worlds/sc2/__init__.py

```diff
@@ -0,0 +1,482 @@
+from .MissionTables import MissionInfo, SC2Campaign, lookup_name_to_mission, SC2Mission, \
+    SC2Race
+
+
+class Starcraft2WebWorld(WebWorld):
+    setup = Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up the Starcraft 2 randomizer connected to an Archipelago Multiworld",
+        "English",
+        "setup_en.md",
+        "setup/en",
+        ["TheCondor", "Phaneros"]
+    )
+
+    tutorials = [setup]
+
+
+class SC2World(World):
+    """
+    StarCraft II is a science fiction real-time strategy video game developed and published by Blizzard Entertainment.
+    Play as one of three factions across four campaigns in a battle for supremacy of the Koprulu Sector.
+    """
+
+    game = "Starcraft 2"
+    web = Starcraft2WebWorld()
+    data_version = 6
+
+    item_name_to_id = {name: data.code for name, data in get_full_item_list().items()}
+    location_name_to_id = {location.name: location.code for location in get_locations(None, None)}
+    options_dataclass = Starcraft2Options
+    options: Starcraft2Options
+
+    item_name_groups = item_name_groups
+    locked_locations: typing.List[str]
+    location_cache: typing.List[Location]
+    mission_req_table: Dict[SC2Campaign, Dict[str, MissionInfo]] = {}
+    final_mission_id: int
+    victory_item: str
+    required_client_version = 0, 4, 3
+
+    def __init__(self, multiworld: MultiWorld, player: int):
+        super(SC2World, self).__init__(multiworld, player)
+        self.location_cache = []
+        self.locked_locations = []
+
+    def create_item(self, name: str) -> Item:
+        data = get_full_item_list()[name]
+        return StarcraftItem(name, data.classification, data.code, self.player)
+
+    def create_regions(self):
+        self.mission_req_table, self.final_mission_id, self.victory_item = create_regions(
+            self.multiworld, self.player, get_locations(self.multiworld, self.player), self.location_cache
+        )
+
+    def create_items(self):
+        setup_events(self.player, self.locked_locations, self.location_cache)
+
+        excluded_items = get_excluded_items(self.multiworld, self.player)
+
+        starter_items = assign_starter_items(self.multiworld, self.player, excluded_items, self.locked_locations, self.location_cache)
+
+        fill_resource_locations(self.multiworld, self.player, self.locked_locations, self.location_cache)
+
+        pool = get_item_pool(self.multiworld, self.player, self.mission_req_table, starter_items, excluded_items, self.location_cache)
+
+        fill_item_pool_with_dummy_items(self, self.multiworld, self.player, self.locked_locations, self.location_cache, pool)
+
+        self.multiworld.itempool += pool
+
+    def set_rules(self):
+        self.multiworld.completion_condition[self.player] = lambda state: state.has(self.victory_item, self.player)
+
+    def get_filler_item_name(self) -> str:
+        return self.multiworld.random.choice(filler_items)
```

Comment:
```suggestion
        return self.random.choice(filler_items)
```

Would recommend switching from `multiworld.random` to `self.random` or `world.random`, both here and wherever else you are using `multiworld.random`.
To keep it brief, it makes the randomness more deterministic in most cases.

Reply:
`multiworld.random` is used often in `PoolFilter.py`. This approach would only deal with the `__init__.py` instances

Reply:
Hmmm, it definitely should still be changed for those other instances as well. Let me see real quick what's going on

Reply:
Ok, two things

You could replace those instances with `multiworld.worlds[player].random`

However, I in general do not think it is a good paradigm to pass `(multiworld, player)` to your functions. If you don't wanna change everything now, then that's fine, there's the easy fix I just mentioned. But the better way is to pass `world: SC2World` to your functions and then get multiworld through `world.multiworld` and player through `world.player` if you need them.
There is some weirdness with type hints if you take that approach, but it is generally better in my opinion.

Reply:
So instead `multiworld.random` the better approach is `multiworld[player].random`?

Reply:
`multiworld.worlds[player].random` is better than what is in place now, yes

Reply:
`multiworld.worlds[player].random`, or just `world.random` if you refactor to pass SC2World instead of multiworld and player

Reply:
Switched to use `world.random` haven't changed the functions APIs as that'd surely need more refactoring elsewhere

Reply:
Tested with new generations -> success

Reply:
Looked into that and some areas already hard-require the player parameter (creating Locations, etc) so getting the world as `multiworld.worlds[player]` shouldn't hurt